packages/integration-tests/src/entities/Publisher.ts

```diff
@@ -8,12 +8,18 @@ export class Publisher extends PublisherCodegen {
     load: (entity) => entity.populate(allImagesHint),
     get: (entity) => {
       const loaded = entity as Loaded<Publisher, typeof allImagesHint>;
-      const images = [...loaded.images.get] as (Image | undefined)[];
-      loaded.authors.get.forEach((author) => {
-        images.push(author.image.get);
-        author.books.get.forEach((book) => images.push(book.image.get));
-      });
-      return images.filter((imageOrUndefined) => imageOrUndefined !== undefined) as Image[];
+
+      return loaded.authors.get
+        .reduce(
+          (images, author) => {
+            images.push(author.image.get!);
+            author.books.get.forEach((book) => images.push(book.image.get!));
+            return images;
+          },
+          [...loaded.images.get],
+        )
+        .filter((imageOrUndefined) => imageOrUndefined !== undefined)
+        .sort((a, b) => ImageTypes.findByCode(a.type)!.sortOrder - ImageTypes.findByCode(b.type)!.sortOrder);
```

Comment:
I die a little bit each time JS doesn't have `sortBy`.

Reply:
also that the cannonical way to avoid sorting in place is `[...someArray].sort(...)`

Reply:
also that for some reason reduce's arguments are backwards